animation_nodes/nodes/color/set_vertex_color.py

```diff
@@ -1,62 +1,129 @@
+from . c_utils import getLoopColorsFromVertexColors, getLoopColorsFromPolygonColors
+
+modeItems = [
+    ("LOOPS", "Loops", "Set color of every loop vertex", "NONE", 0),
+    ("VERTICES", "Vertices", "Set color of every vertex", "NONE", 1),
+    ("POLYGONS", "Polygons", "Set color of every polgon", "NONE", 2)    
+]
+
+colorLayerIdentifierTypeItems = [
+    ("INDEX", "Index", "Get vertex color layer based on the index", "NONE", 0),
+    ("NAME", "Name", "Get vertex color layer based on the name", "NONE", 1)
+]
 
 class SetVertexColorNode(bpy.types.Node, AnimationNode):
     bl_idname = "an_SetVertexColorNode"
     bl_label = "Set Vertex Color"
     errorHandlingType = "EXCEPTION"
 
-    vertexColorName: StringProperty(name = "Vertex Color Group", default = "Col", update = propertyChanged)
-    checkIfColorIsSet: BoolProperty(default = True)
+    mode: EnumProperty(name = "Mode", default = "LOOPS",
+        items = modeItems, update = AnimationNode.refresh)
+
+    colorLayerIdentifierType: EnumProperty(name = "Color Layer Identifier Type", default = "INDEX",
+        items = colorLayerIdentifierTypeItems, update = AnimationNode.refresh)
+
+    useColorList: VectorizedSocket.newProperty()
 
     def create(self):
-        self.newInput("Object", "Object", "object").defaultDrawType = "PROPERTY_ONLY"
-        self.newInput("Color", "Color", "color").defaultDrawType = "PROPERTY_ONLY"
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+
+        if self.colorLayerIdentifierType == "INDEX":
+            self.newInput("Integer", "Color Index", "colorLayerIndex")
+        elif self.colorLayerIdentifierType == "NAME":
+            self.newInput("Text", "Name", "colorLayerName")
+
+        self.newInput(VectorizedSocket("Color", "useColorList",
+            ("Color", "color"), ("Colors", "colors")))
+
         self.newOutput("Object", "Object", "outObject")
 
     def draw(self, layout):
-        layout.prop(self, "vertexColorName", text = "", icon = "GROUP_VCOL")
+        layout.prop(self, "mode", text = "")
 
     def drawAdvanced(self, layout):
-        layout.prop(self, "checkIfColorIsSet", text = "Check Color")
+        layout.prop(self, "colorLayerIdentifierType", text = "Type")
 
-    def execute(self, object, color):
-        if object is None: return object
-        if object.type != "MESH": return object
-        if object.mode == "EDIT":
-            self.raiseErrorMessage("Object is in edit mode")
-            return object
+    def getExecutionFunctionName(self):
+        if self.useColorList:
+            return "execute_ColorsList"
+        else:
+            return "execute_SingleColor"
+
+    def execute_SingleColor(self, object, identifier, color):
+        if object is None: return
+        colorLayer = self.getVertexColorLayer(object, identifier)
 
-        mesh = object.data
+        colorsList = VirtualColorList.create(color, 0).materialize(len(colorLayer.data))
```

Comment:
The second argument of `create` should be a color not a number.

Reply:
Is this correct way?
```
color = Color((0., 0., 0., 1.))
colorsList = VirtualColorList.create(colors, color).materialize(len(colorLayer.data))
```

Reply:
Yes. But name the variable `defaultColor` and lose the point in the arguments, Python doesn't really care.